pulser/simulation/simulation.py

```diff
@@ -325,55 +463,125 @@ def get_hamiltonian(self, time: float) -> qutip.Qobj:
                              "greater than or equal to 0.")
         return self._hamiltonian(time/1000)  # Creates new Qutip.Qobj
 
+    def _reset_samples(self) -> None:
+        """Sets all samples to their default values.
+
+        Used when running several noisy runs, each one setting new samples.
+        """
+        self.samples = {addr: {basis: {}
+                               for basis in ['ground-rydberg', 'digital']}
+                        for addr in ['Global', 'Local']}
+
+    def _init_dephasing(self) -> None:
+        """Initializes dephasing collapse operators."""
+        if self.basis_name == 'digital' or self.basis_name == 'all':
+            raise ValueError("Cannot include dephasing noise in digital-"
+                             "or all-basis.")
+        self.dephasing_prob = 0.05  # Probability of phase (Z) flip
+        self._collapse_ops += [
+            np.sqrt(1 - self.dephasing_prob) * self.op_matrix['I'],
+            np.sqrt(self.dephasing_prob) * (self.op_matrix['sigma_rr']
+                                            - self.op_matrix['sigma_gg'])]
+
+    def _build_hamiltonian_from_seq(self) -> None:
+        """Builds the hamiltonian from sequence samples.
+
+        Updates the new sequence samples that may have changed due to noise,
+        builds default operators and builds the hamiltonian from those
+        coefficients and operators.
+        """
+        self._reset_samples()
+        self._extract_samples()
+        self._build_basis_and_op_matrices()
+        self._construct_hamiltonian()
+
     # Run Simulation Evolution using Qutip
-    def run(self,
-            initial_state: Optional[Union[np.ndarray, qutip.Qobj]] = None,
-            progress_bar: Optional[bool] = None,
+    def run(self, progress_bar: Optional[bool] = None,
             **options: qutip.solver.Options) -> SimulationResults:
-        """Simulates the sequence using QuTiP's solvers.
+        """Simulate the sequence using QuTiP's solvers.
 
-        Keyword Args:
-            initial_state (array): The initial quantum state of the
-                evolution. Will be transformed into a ``qutip.Qobj`` instance.
-            progress_bar (bool): If True, the progress bar of QuTiP's
-                ``qutip.sesolve()`` will be shown.
+        Will return NoisyResults if it detects any noise in the SimConfig.
+        Otherwise will return CleanResults.
```

Comment:
Is all noise incoherent?

Reply:
We chose by design to implement noises with the Counter approach, because it makes sense when averaging results. What's more, you can't see, for example, Doppler damping, without actually averaging the results over many runs (we tried this with a "clean" quantum-state with @sebgrijalva and it didn't show any damping).